detail.html

```diff
@@ -0,0 +1,34 @@
+<%inherit file="app_base.htm" />
+
+<%block name ="content_top">
+</%block>
+
+<div id = "pagestuff">
+  <%block name = "content_middle">
+    <div id = "small_imgs">
+    <ul>
+      %for pic in product.image_urls():
+        <li class = "hover_item"><img src = "${pic}" alt = "main picture for ${product_name}"/></li>
+      %endfor
+    </ul>
+    </div>
+    <div id = "big_img">
+      <img src="${product.image_url()}" alt = "alternate picture for ${product_name}"/>
+
+    </div>
+    <div id = "detail_desc">
+    <h1>${product_name}</h1>
+
+      <p class = "prod_desc">${product.description}</p>
+    </div>
+
+    %if product.TITLE == "Bulk":
```

Comment:
I didn't know you can do this! I used hasattr

Reply:
hasattr() is a really cute way to do it.

Reply:
I didn't know that even existed! Good to know